src/main/java/com/aws/greengrass/deployment/IotJobsHelper.java

```diff
@@ -268,17 +275,45 @@ private static void unwrapExecutionException(ExecutionException e)
     }
 
     @Override
-    @SuppressFBWarnings
+    @SuppressWarnings("PMD.CollapsibleIfStatements")
     public void postInject() {
+        deviceConfiguration.onAnyChange((what, node) -> {
+            if (node != null && what.equals(WhatHappened.childChanged)) {
+                // List of configuration nodes that we need to reconfigure for if they change
+                if (node.childOf(DEVICE_PARAM_THING_NAME) || node.childOf(DEVICE_PARAM_IOT_DATA_ENDPOINT)
+                        || node.childOf(DEVICE_PARAM_PRIVATE_KEY_PATH)
+                        || node.childOf(DEVICE_PARAM_CERTIFICATE_FILE_PATH) || node.childOf(DEVICE_PARAM_ROOT_CA_PATH)
+                        || node.childOf(DEVICE_PARAM_AWS_REGION)) {
+                    try {
+                        validateAndSetupCommWithIotJobs(deviceConfiguration);
+                    } catch (DeviceConfigurationException e) {
+                        logger.atWarn().log("Device not configured to talk to AWS Iot cloud. "
+                                + "IOT job deployment is offline: {}", e.getMessage());
+                        return;
+                    }
+                }
+            }
+        });
+
         try {
-            // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error message to user
-            deviceConfiguration.validate();
+            validateAndSetupCommWithIotJobs(deviceConfiguration);
         } catch (DeviceConfigurationException e) {
             logger.atWarn().log("Device not configured to talk to AWS Iot cloud. IOT job deployment is offline: {}",
                     e.getMessage());
             return;
         }
+    }
+
+    private void validateAndSetupCommWithIotJobs(DeviceConfiguration deviceConfiguration)
+            throws DeviceConfigurationException {
 
+        // Not using isDeviceConfiguredToTalkToCloud() in order to provide the detailed error message to user
+        deviceConfiguration.validate();
+        setupCommWithIotJobs();
+    }
+
+    @SuppressFBWarnings
```

Comment:
Which warning is this? Could we avoid suppressing warnings unless it's a false alarm?

Reply:
It was coming because of ignoring return value of executorService.submit(). It exists in current code in main as well. Refactored it and removed the warning.

Reply:
Then could you use `executorService.execute()` instead of `submit()`? Even if it's done this way in existing code we should avoid these patterns if we can.

Reply:
I tried that and it lead to some failures in tests. I did not debug it then. I will make the change and debug it now.